activesupport/lib/active_support/core_ext/module/attribute_accessors_per_thread.rb

```diff
@@ -118,6 +118,8 @@ def #{sym}=(obj)
   #   class Customer < Account
   #   end
   #
+  #   Account.user = 'DHH'
```

Comment:
This was set just a few lines up. If we aren't willing to accept that as context, we don't have an Account class to inherit.